packages/node/src/services.ts

```diff
@@ -202,3 +204,99 @@ export function confirmLocalFirebaseConfigurationEnvVars() {
     );
   }
 }
+
+export interface PostgresConfiguration {
+  user: string;
+  host: string;
+  database: string;
+  password: string;
+  port: number;
+}
+
+export const POSTGRES_CONFIGURATION_ENV_KEYS = {
+  user: "POSTGRES_USER",
+  host: "POSTGRES_HOST",
+  database: "POSTGRES_DATABASE",
+  password: "POSTGRES_PASSWORD",
+  port: "POSTGRES_PORT"
+};
+
+export const EMPTY_POSTGRES_CONFIG: PostgresConfiguration = {
+  user: "",
+  host: "",
+  database: "",
+  password: "",
+  port: 0
+};
+
+export class PostgresServiceFactory {
+  private client: Client;
+
+  constructor(configuration: PostgresConfiguration) {
+    this.client = new Client(configuration);
+  }
+
+  static async connect(host: string, port: number) {
+    new PostgresServiceFactory({
+      ...EMPTY_POSTGRES_CONFIG,
+      host,
+      port
+    });
+  }
+
+  async createStoreService(storeServiceKey: string): Promise<IStoreService> {
+    await this.client.connect();
+    console.log("Connected to Postgres");
+    return new PostgresStoreService(this.client, storeServiceKey);
+  }
+}
+
+class PostgresStoreService implements IStoreService {
+  constructor(
+    private readonly pgClient: Client,
+    private readonly storeServiceKey: string
+  ) {}
+
+  async get(key: string): Promise<any> {
+    const result = await this.pgClient.query(
+      SQL`SELECT * FROM "`
+        .append(this.storeServiceKey)
+        .append(SQL`" WHERE "key" = ${key}`)
+    );
+    return result.rows[0] && result.rows[0].value[key];
+  }
+
+  async set(pairs: { key: string; value: any }[]): Promise<any> {
```

Comment:
@rhlsthrm can we make this a transaction across the set of pairs that's being written?

Seems like the current implementation loses atomicity which could potentially cause state dissonance.

Reply:
This is a transaction, that's what the `BEGIN` and `COMMIT` queries are for. The query will be atomic because of this.

Reply:
Ah but I still need to catch the error and `ROLLBACK`. I'll address this.

Reply:
Eh I should have been more specific. I meant across all pairs, instead of doing the write per iteration of the loop.

Similar to this: https://github.com/counterfactual/monorepo/blob/master/packages/node/src/services.ts#L180